CMake/FindPETSc.cmake

```diff
@@ -14,26 +14,37 @@
 
 # Find the PETSc library
 #
-# Authors : D. THOMAS
+# Authors : D. THOMAS & co
 #
 # This will define
 #
-#  PETSC_DIR                  - where to find petsc4py/petsc4py.i, etc.
-#  PETSC4PY_FOUND             - TRUE if PETSc is found
+#  <PETSC_DIR>                - root folder of petsc
 #  PETSC_INCLUDE_PATH         - where to find petsc.h
 #  PETSC_LIBRARIES            - path to libpetsc.so
 
-FIND_PATH(PETSC_DIR include/petsc.h HINTS ENV PETSC_DIR PATHS /usr/local/lib/python2.7/dist-packages/petsc /opt/local/lib/petsc $ENV{HOME}/petsc $ENV{PETSC_DIR})
+FIND_PATH(PETSC_DIR include/petsc.h HINTS ENV PETSC_DIR PATHS /opt/local/lib/petsc $ENV{HOME}/petsc $ENV{PETSC_DIR})
 
-FIND_LIBRARY(PETSC_LIBRARIES NAMES petsc PATHS ${PETSC_DIR}/lib  ${PETSC_DIR}/$ENV{PETSC_ARCH}/lib)
+IF (PETSC_DIR)
+  FIND_LIBRARY(PETSC_LIBRARIES NAMES petsc PATHS ${PETSC_DIR}/lib  ${PETSC_DIR}/$ENV{PETSC_ARCH}/lib)
```

Comment:
If nobody has huge modifications in their own branch once this PR is merged, I think we could make an additional PR just after this one fixing tabs, spaces, indentation, upper/lowercase chars, unicode.